pkg/agent/plugin/svidstore/disk/disk.go

```diff
@@ -228,35 +311,69 @@ func newDiskStore(metaData []string, dir string) (*diskStore, error) {
 		return nil, err
 	}
 
+	if err := sameDirectory(certChainFilePath, keyFilePath); err != nil {
+		return nil, err
+	}
+
 	bundleFilePath, err := getFileMetadata(metadataMap, "bundlefile")
 	if err != nil {
 		return nil, err
 	}
 
+	if err := sameDirectory(keyFilePath, bundleFilePath); err != nil {
+		return nil, err
+	}
+
+	var gidInt int64
+
+	gid := metadataMap["gid"]
+	groupName := metadataMap["group"]
+
+	if gid != "" {
```

Comment:
I think it can be simplified with a switch:
```
swtich {
   case gid != "" && groupName != "":
       return nil, status.Error(codes.InvalidArgument, "either gid or group must be specified, not both")
   case gid != "":
        ....
    case groupName != "" :
        .....
}
```